src/js/entities-service/forms.js

```diff
@@ -14,7 +14,11 @@ const Entity = BaseEntity.extend({
     'fetch:forms:byAction': 'fetchByAction',
     'fetch:forms:definition:byAction': 'fetchDefinitionByAction',
   },
-  fetchDefinition(formId) {
+  fetchDefinition(formId, actionId) {
+    if (actionId) {
+      return this.fetchDefinitionByAction(actionId);
+    }
```

Comment:
Would it make more sense to keep this method as-is, and switch to `fetchDefinitionByAction` with outside calls, rather than having to modify every call to `fetchDefinition`?

Reply:
nope.. this got a lot more complicated than I expected with the other entity.  we'll have to split the service now because for patient sidebar widgets are not really going to be using the form revisions and so the actions will always use the revision and the patient/preview will always use the specific form.id.  Might have to wait until the following release.